packages/ember-glimmer/tests/integration/syntax/if-unless-test.js

```diff
@@ -24,3 +28,85 @@ moduleFor('Syntax test: {{#if}} and {{#unless}} without inverse', class extends
   }
 
 });
+
+moduleFor('Syntax test: {{#if}}', class extends RenderingTest {
+
+  ['@test child conditional should not render children if parent conditional becomes false'](assert) {
+    let childCreated = false;
+
+    this.registerComponent('foo-bar', {
+      template: 'foo-bar',
+      ComponentClass: Component.extend({
+        init() {
+          this._super(...arguments);
+          childCreated = true;
+        }
+      })
+    });
+
+    this.render('{{#if cond1}}{{#if cond2}}{{foo-bar}}{{/if}}{{/if}}', { cond1: true, cond2: false });
+
+    assert.ok(!childCreated);
+    this.assertText('');
+
+    this.runTask(() => this.rerender());
+
+    assert.ok(!childCreated);
+    this.assertText('');
+
+    this.runTask(() => {
+      set(this.context, 'cond2', true);
+      set(this.context, 'cond1', false);
+    });
+
+    assert.ok(!childCreated);
+    this.assertText('');
+
+    this.runTask(() => {
+      set(this.context, 'cond2', false);
+      set(this.context, 'cond1', true);
+    });
+
+    assert.ok(!childCreated);
+    this.assertText('');
+  }
+
+  ['@test using `if` with an `{{each}}` destroys components when transitioning to and from inverse (GH #12267)'](assert) {
+    let destroyedChildrenCount = 0;
+
+    this.registerComponent('foo-bar', {
+      template: '{{number}}',
+      ComponentClass: Component.extend({
+        willDestroy() {
+          this._super();
+          destroyedChildrenCount++;
+        }
+      })
+    });
+
+    this.render(`
```

Comment:
We can use the `strip` helper instead of using `{{~...~}}`, see https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/integration/application/rendering-test.js#L23-L29